### PR TITLE
feat: add new design for errors on deposit modal [WEB-1466] 

### PR DESCRIPTION
**Pull request by FiboApe**
Screenshots:
Screenshot 2022-04-21 at 20 07 07
Screenshot 2022-04-22 at 13 42 36
Screenshot 2022-04-22 at 13 42 58

### CODE REVIEW

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (karelianpie) [inline src/client/components/app/Transactions/components/TxError.tsx:105]: This might lead to unexpected results because this is asynchronous, we should use the previous state rather setting this directly

```suggestion
            onClick={() => setIsExpanded((prevIsExpanded) => !prevIsExpanded)}
```

File: src/client/themes/cyberpunk/index.ts
Comment (karelianpie) [inline src/client/themes/cyberpunk/index.ts:73]: Is this the background-color? I think if so, we should try and use `backgroundColor` instead, unsure if that's a big refactor

File: src/utils/validations.ts
Comment (karelianpie) [inline src/utils/validations.ts:114]: I feel these errors could be a bit more meaningful if we were to use the logic to get here. Example:
```suggestion
    return { error: `Invalid amount: ${amountInWei}` };
```

File: src/client/components/app/Transactions/components/TxContainer.tsx
Comment (turtlemoji) [inline src/client/components/app/Transactions/components/TxContainer.tsx:65]: This will cause rerendering and wrong style in most of the txcontainers, we should leave it or change the height to another value to reflect the new changes and don't cause an overflow

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (turtlemoji) [inline src/client/components/app/Transactions/components/TxError.tsx:9]: we just do fill: currentColor, and color: inherit, so you will not need to specify it again here and only on the parent

File: src/client/themes/dark/index.ts
Comment (turtlemoji) [inline src/client/themes/dark/index.ts:110]: we have this too, as global vars in default file. Don't know if it should change between themes or not, I will leave it at your criteria

Comment (turtlemoji) [reply src/client/themes/dark/index.ts:110]: image

File: src/utils/validations.ts
Comment (turtlemoji) [inline src/utils/validations.ts:118]: we should create another PR in the future and have error codes or something like "INSUFFICIENT_FUNDS" so we can just translate them and have them all in a .json file

**Comment by turtlemoji** [comment]
@FiboApe can you add another error component in the test modal with description so we can try the expand more easily?
image

File: src/utils/validations.ts
Comment (FiboApe) [reply src/utils/validations.ts:114]: in this case amountInWei would be just what the user set on the input, I don't know if it makes much sense to tell the user that

File: src/client/components/app/Transactions/components/TxContainer.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/components/TxContainer.tsx:65]: Height should be defined by the content and not the txcontainer (maybe we can set a max one), which of the other contents would make this break?

File: src/client/themes/dark/index.ts
Comment (FiboApe) [reply src/client/themes/dark/index.ts:110]: But those are for alerts, this is the modal, should we use the same for both?

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/components/TxError.tsx]: fixed on 43649ef98fa2d6e9d704f21f6f7892a3ba13f7f2

File: src/client/themes/cyberpunk/index.ts
Comment (FiboApe) [reply src/client/themes/cyberpunk/index.ts]: fixed on 43649ef98fa2d6e9d704f21f6f7892a3ba13f7f2

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/components/TxError.tsx]: solved on 5c0f6e767314e1ad4f1dc71847736db858f9970f

File: src/client/components/app/Transactions/components/TxContainer.tsx
Comment (turtlemoji) [reply src/client/components/app/Transactions/components/TxContainer.tsx:65]: In the designs the height was fixed, and it was not defined by the content in all cases. More like having a max and min height. If not, it will overflow in the vault details, or cause rerenders

File: src/client/themes/dark/index.ts
Comment (turtlemoji) [reply src/client/themes/dark/index.ts:110]: Idk, depending if the colors are ok, in reality that colors are just global variables, not only for alerts itself. It depends if the error colors should change betweeen themes I suppose

**Comment by turtlemoji** [comment]
@FiboApe should we create another PR for the transaction messages? (Success, loading, etc)
image

**Comment by turtlemoji** [comment]
@FiboApe can you bump package minor version?

File: src/client/components/app/Transactions/components/TxContainer.tsx
Comment (turtlemoji) [reply src/client/components/app/Transactions/components/TxContainer.tsx:65]: image
In this case, for exxample, we should be flexing the content to the bottom to occupy all the remaining space

File: src/utils/validations.ts
Comment (karelianpie) [reply src/utils/validations.ts:114]: Good point 😅 I forgot these were coming from there. All good, I had a look at the others and the user should be able to deduce why the error is happening. Resolving!

File: src/client/components/app/Transactions/TestTx.tsx
Comment (karelianpie) [inline src/client/components/app/Transactions/TestTx.tsx:15]: What's this?

**Review by karelianpie — APPROVED** [review]
Is it me or this would be nicer with a little bit more padding around the error message? Text seems a bit too close to the edge

feat__add_new_design_for_errors_on_deposit_modal__WEB-1466__by_FiboApe_·_Pull_Request__607_·_yearn_yearn-finance-v3

Otherwise, LGTM. Thanks for adding the screenshots! 👍

**Comment by turtlemoji** [comment]
> Is it just me or this would be nicer with a little bit more padding around the error message?
> 
> Text seems a bit too close to the edge:
> 
> feat__add_new_design_for_errors_on_deposit_modal__WEB-1466__by_FiboApe_·_Pull_Request__607_·_yearn_yearn-finance-v3
> 
> Otherwise, LGTM. Thanks for adding the screenshots! 👍

image
 
 This is what design delivered so I will follow it for now just to don't stall waiting for their response, lgtm

**Comment by karelianpie** [comment]
feat__add_new_design_for_errors_on_deposit_modal__WEB-1466__by_FiboApe_·_Pull_Request__607_·_yearn_yearn-finance-v3

@turtlemoji when I overlay it appears the design has a bit more padding at the top at least, looks like another 8px perhaps

**Review by karelianpie — CHANGES_REQUESTED** [review]
Sorry, I think we should be closer to the designs, the error message needs a bit more breathing room

**Comment by turtlemoji** [comment]
Yes, if is different lets change it :) @FiboApe

**Comment by turtlemoji** [comment]
ok, matched design with the figma, (but removed .8 of margin-top because it looked really bad in the implementation, figma used 1.6rem for margin-top and I left only .8rem)
Moved click to all the title so it works on mobile (too difficult to click the dropdown icon)
Also changed the icon to a new one used in the figma (yeah, it is different)

67c055b5b68004d474b2dac59d2b6c8f4668c9e0

**Comment by turtlemoji** [comment]
image

**Comment by turtlemoji** [comment]
Also @FiboApe , I think we will have to change the colors for some themes:
image

**Comment by karelianpie** [comment]
> ok, matched design with the figma, (but removed .8 of margin-top because it looked really bad in the implementation, figma used 1.6rem for margin-top and I left only .8rem)

Fair enough, thanks! I think it looks good now 🎉 

> Moved click to all the title so it works on mobile (too difficult to click the dropdown icon)

Good idea! 👍

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (karelianpie) [inline src/client/components/app/Transactions/components/TxError.tsx:18]: nit: Is this how lint/prettier prefers it? Looks quite hard to read

**Review by karelianpie — APPROVED** [review]

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (turtlemoji) [reply src/client/components/app/Transactions/components/TxError.tsx:18]: Yeah. 
I would love this
image

But prettier gets really angry at that.
Maybe @FiboApe can create/edit a rule for this? He is really good with lint rules

**Review by turtlemoji — CHANGES_REQUESTED** [review]
fix cyberpunk colors of txError

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (karelianpie) [reply src/client/components/app/Transactions/components/TxError.tsx:18]: 100%, that would be the way to format that in a much more readable way

File: src/client/themes/dark/index.ts
Comment (FoxTheSin) [reply src/client/themes/dark/index.ts:110]: For ease sake, keep it the same. After release (because I already freaking love this update to the errors) we can update UI for cleaner/easier reading. Function first IMO.

File: src/client/components/app/Transactions/TestTx.tsx
Comment (FiboApe) [inline src/client/components/app/Transactions/TestTx.tsx:16]: this was actually formatted by prettier

Comment (turtlemoji) [reply src/client/components/app/Transactions/TestTx.tsx:16]: Yeah, idk why it break the build in one stage

**Review by karelianpie — APPROVED** [review]

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (turtlemoji) [inline src/client/components/app/Transactions/components/TxError.tsx:91]: typo

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (turtlemoji) [inline src/client/components/app/Transactions/components/TxError.tsx:104]: typo?

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (turtlemoji) [inline src/client/components/app/Transactions/components/TxError.tsx:126]:
```suggestion
    // NOTE Clean state once error message changes
```

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/components/TxError.tsx:104]: nope, it adds 80% for opacity

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/components/TxError.tsx:91]: nope, it adds 80% for opacity

**Comment by FiboApe** [comment]
@turtlemoji fixed height issue and allowed it to have variable height here: e6ee4b6b92d8cf2ca25b28ae29ab1425fd69798d
Screenshot 2022-04-28 at 14 39 01
Screenshot 2022-04-28 at 14 39 13
Screenshot 2022-04-28 at 14 39 21
Screenshot 2022-04-28 at 14 39 27

**Review by karelianpie — APPROVED** [review]

File: src/client/components/common/Tabs.tsx
Comment (turtlemoji) [inline src/client/components/common/Tabs.tsx:106]: @FiboApe missing space after `;`. I like to not use the ternary tho and just do more lines in this case

**Review by turtlemoji — APPROVED** [review]

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (turtlemoji) [reply src/client/components/app/Transactions/components/TxError.tsx:91]: the thing is, what appens if the variable is an RGB, it will break, this only works with hexas

Comment (xgambitox) [reply src/client/components/app/Transactions/components/TxError.tsx:91]: should we rather set the opacity separately ?

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (xgambitox) [reply src/client/components/app/Transactions/components/TxError.tsx:104]: should we rather set the opacity separately ?

Comment (xgambitox) [reply src/client/components/app/Transactions/components/TxError.tsx:104]: should we rather set the opacity separately ?

**Review by xgambitox — APPROVED** [review]

**Review by karelianpie — APPROVED** [review]

File: src/client/components/app/Transactions/components/TxError.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/components/TxError.tsx:91]: I can add a new constant to the theme just to have it as opacity, but there is no way to set opacity of the background separately

Comment (turtlemoji) [reply src/client/components/app/Transactions/components/TxError.tsx:91]: you can always create a function to check if the color is an RGB, RGBA, or an hexa, and then apply the opacity depending that lol. 
I would just not use opacity in this case and not have problems in the future, and just try with another var. like mixing background color and surface color.

**Pull request merged**
